src/Symfony/Component/Validator/Constraints/BackedEnumValueValidator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * BackedEnumValueValidator validates that a backed enum case can be hydrated from a value.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class BackedEnumValueValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof BackedEnumValue) {
+            throw new UnexpectedTypeException($constraint, BackedEnumValue::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        try {
```

Comment:
```suggestion
        if ($value instanceof \Stringable) {
            $value = (string) $value;
        }

        if (!\is_string($value)) {
            throw new UnexpectedValueException($value, 'string');
        }

        try {
```

Reply:
backed enums can be ints, im not sure it's covered in tests

Reply:
`int` enum case is covered in tests : https://github.com/symfony/symfony/blob/b0a1c635f9fe02f0f585d330b7e897a4f5b364c0/src/Symfony/Component/Validator/Tests/Constraints/BackedEnumValueValidatorTest.php#L141

Reply:
Yes, but @xabbuh's proposal wouldn't work for integers. I think that's what @ro0NL was getting at. 🙂

Reply:
If checking the type would free us from catching the `TypeError`, I'd say go for it. But given that we still need to catch it, I would not got with @xabbuh's proposal.

Reply:
So we catch the `TypeError` and upcast it?

Reply:
yes, we could differentiate between `TypeError` and `ValueError`, respectively translating to `UnexpectedValueException` and a user constraint error in symfony

but then we should use `BackedEnum::from()` AFAIK, because `tryFrom` seems really quirky: https://3v4l.org/54qFs

Reply:
Sorry but I don't understand what this change will improve... Can someone explain to me please ?

Reply:
@xabbuh do you agree to resolve this conversation or do you want to discuss about it ?